containers/runtime/xds/config/README.md

```diff
@@ -0,0 +1,33 @@
+* TTL: Optional fields to set TTL for each resource item.
+* Items: Items are maps within each resource type, the maps' keys are the individual resource names and the values are the actual resources with an optional TTL
+* VersionMap: VersionMap holds the current hash map of all resources in the snapshot. This field should remain nil until it is used. In our use case, we only unmarshal the configuration into Snapshot, the VersionMap remains nil. VersionMap is only to be used with delta xDS.
+
+The source of truth for the fields within each resource in the resource map are their protos.
+The 8 types of resources:
+
+* Listener: [config.listener.v3.Listener proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/config/listener/v3/listener.proto#L39)
+* Endpoint: [config.endpoint.v3.ClusterLoadAssignment proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/config/endpoint/v3/endpoint.proto#L33)
+* Cluster: [config.cluster.v3.Cluster proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/config/cluster/v3/cluster.proto#L47)
+* Route: [config.route.v3.RouteConfiguration proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/config/route/v3/route.proto#L26)
+* ScopedRoute: [config.route.v3.ScopedRouteConfiguration proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/config/route/v3/scoped_route.proto#L83)
+* ExtensionConfig: [config.core.v3.ExtensionConfigSource proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/config/core/v3/extension.proto#L47)
+* Secret: [extensions.transport_sockets.tls.v3.Secret proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/extensions/transport_sockets/tls/v3/secret.proto#L40)
+* Runtime: [service.runtime.v3.Runtime proto](https://github.com/envoyproxy/envoy/blob/3865cbaec08c5ce44b439eb77e46fe866e37e81a/api/envoy/service/runtime/v3/rtds.proto#L50)
+
+## Default configuration JSON file
+
+The daufult_config.json file contains a piece of configuration that have two listeners with name: `default_testEnvoyListenerName` and `default_testGrpcListenerName`, one Cluster resource, one Route resource and one Endpoint resource. The two listeners are pointing to the same Cluster, eventually the same Endpoint resource.
+
+## User supplied configuration JSON file
+
+If user wish to alter the default configuration, a user defined configuration can be created and once the path of the user supplied configuration file is provided, the xDS server then server the user defined configuration. User supplied configurations are updated on top of the default configuration, so U=user only need to supply the part that wish to alter. If the user did not supply any configuration, the default configuration will be used for xDS server.
```

Comment:
Can you please document how the user can supply their own configuration? Is there a fixed path? CLI flag? Environment variable?

Reply:
Actually there is not much to do to use a user defined configuration, the only thing is to create a configuration JSON file just like the default configuration in the config directory. At the time starting an xDS server, there is a flag that can be used to supply the path of the user defined configuration file. 
https://github.com/grpc/test-infra/blob/51516f03ee316e50454a630558c9d2adc5fe11bd/containers/runtime/xds/cmd/main.go#L41

It is a part of the https://github.com/grpc/test-infra/pull/235, I am not so sure about the flag name yet, I can mentioned it here and will update that if needed. 

I will also come up with more explanation on how to use flags in xDS server's README.

Reply:
> create a configuration JSON file just like the default configuration in the config directory

Right. So where is the config directory? In general, what I'm trying to get at is that a newcomer should be able to read the README and know everything they need to supply a custom xDS configuration.

> It is a part of the #235, I am not so sure about the flag name yet, I can mentioned it here and will update that if needed.
Sure. If the feature isn't completed by this PR, we don't need to hold it up for this. We just need to make sure that the README gets to that state at _some_ point.